### ISSUE TITLE
Update IPHONEOS_DEPLOYMENT_TARGET from 10.0 to 11.0 in 3.5.1 and greater

### ISSUE THREAD

**Issue by biblical-text** [godotengine/godot#68720]
### Godot version

3.5.1

### System information

n/a

### Issue description

The generated iOS project build files sets IPHONEOS_DEPLOYMENT_TARGET as 10.0, which is invalid in the latest versions of Xcode. Recommend 11.0 is set as default.

### Steps to reproduce

  Export any hello world project to iOS

### Minimal reproduction project

Export any hello world project to iOS

**Comment by clayjohn** [comment]
cc @bruvzg

**Comment by Calinou** [comment]
I'm surprised Xcode eventually blocksyou from using old iOS versions as a baseline... iOS 10 should still be able to run Godot 3.x projects.

**Comment by zaddok** [comment]
I can confirm Xcode 14.1 warns 10.0 is an invalid setting, although it still allows the application to build.

**Comment by biblical-text** [comment]
Yes, the exact error Xcode gives is:

`The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 10.0, but the range of supported deployment target versions is 11.0 to 16.1.99.`

**Comment by bruvzg** [comment]
The only officially supported versions are 15 and 16. So dropping support for the iOS 10 is expected, some minor updates for the Xcode 12 did the same for iOS 9.

In the 4.0 it's already set to `11`.

**Comment by hiulit** [comment]
I'm using Godot 3.5.1 with Xcode 14.2 (macOS 12.7.4) and I get the same error.

```
 warning: The iOS deployment target 'IPHONEOS_DEPLOYMENT_TARGET' is set to 10.0, but the range of supported deployment target versions is 11.0 to 16.2.99.
 ```
 
 It's a minor issue, because once you open the Xcode project it opens a pop-up telling you to upgrade some settings, it automatically fixes it for you and that's it, but maybe it would be a good idea to add a new parameter to the Export settings for iOS in Godot? To specify the "(minimum) deployment target" so we don't get that warning?

**Comment by oeleo1** [comment]
Yes, bumping the minimum iOS version is usually taken care of by @bruvzg (Godot's official iOS guru) but there are plenty of minor issues out there lingering around that eventually get fixed for 3.x, like [this one for example](https://github.com/godotengine/godot/pull/89609). As long as it is not blocking nor presents a significant inconvenience - it's okay.

But you are right that it would have been better if the Wolrd we're living in was ideal and we really had 0 warnings and 0 errors on every compile that we do with Xcode, including zero warnings about translation projects and what not. But one eventually accepts the fact that the World we're living in is never perfect so we boringly swallow a few orange Xcode warnings every time we compile a project without bothering to make a big deal out of it. ;-) usually in such cases, you're very welcome to contribute a patch.

**Comment by bruvzg** [comment]
> but maybe it would be a good idea to add a new parameter to the Export settings for iOS in Godot? To specify the "(minimum) deployment target" so we don't get that warning?

It is there in 4.x: https://github.com/godotengine/godot/blob/f6a78f83aa4b74aa5cb80ca2e3419448b1998e4f/platform/ios/export/export_plugin.cpp#L166
Might be worth backporting to 3.6 as well.

Min. version is already bumped to `12.0` for both 4.x and 3.6.

**Comment by hiulit** [comment]
If it's not much work for you @bruvzg , I think it would be a good addition to 3.x.

**Comment by madebydavid** [comment]
This is still present in current `3.x` and Apple has begun sending emails after you upload a game built with 3.6.2:

> **ITMS-90068: MinimumOSVersion too low** - This app has a MinimumOSVersion of 12.0. Starting later 
> this year, all iOS apps must have a MinimumOSVersion of 13.0 or later in order to be uploaded to App 
> Store Connect or submitted for distribution. 
> 
> **ITMS-90068: MinimumOSVersion too low** - This app has a MinimumOSVersion of 12.0. Starting in 
> Spring 2027, all iOS apps must have a MinimumOSVersion of 15.0 or later in order to be uploaded to App 
> Store Connect or submitted for distribution. 

I also believe that Xcode 27 (currently beta) will force a minimum deployment target of iOS 15.0.

Bumping IPHONEOS_DEPLOYMENT_TARGET to 15 (4.x's minimum is [already 15.0](https://github.com/godotengine/godot/pull/121030)) will mean people don't need to manually edit the exported xcode project files.

**Comment by lawnjelly** [comment]
Closed by #122324.
